### PR TITLE
[NEW] Add form components "creator kit"

### PR DESCRIPTION
**Pull request by Dnouv**
This PR introduces a form component.
Here are some preview images.

Thank you!

Note: Please do not merge. WIP

### CODE REVIEW

**Comment by Sing-Li** [comment]
Thanks @Dnouv !    Cool.    

The current workflow fo RC4Community has no interactive "design time" behaviours.   Since your code here entails only interactive design-time behaviours, it is best to have it running standalone as a separate tool.

It can generate the actual JSX , or the source code of the react component to display/manage the form directly.   

Or a better solution is if this tool will write the "form description data" to strapi, and then have a set of custom ReactJS tag that can render this form during build time by fetching the "form description data".

**Review by Sing-Li — CHANGES_REQUESTED** [review]
As per our conversation on Rocket.Chat @Dnouv 

1) please place your creation tool as an INDEPENDENT TOOL under `/contrib/formcreator` directory, it should be a standalone 100% independent nodeJS based tool - and not part of RC4Community framework

2) please create components under `components` directory that will be re-usable and renders the form for the community builder

3) please create a demo page under `pages/formdemo` that shows FOUR or FIVE instances of your forms components - displaying FOUR or FIVE forms simultaneously on one single page (nothing else) ... this will illustrate to everyone that your component is easily re-usable

Other than the above, please do not add any additional source files that may alter the structure of the project

File: app/pages/forms/formdemo.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+function FormDemo({ formFields }) {
+  return (
+    <div>
+      <Head>
+        <title>Form</title>
+        <meta name="description" content="Rocket.Chat form tool demo" />
+        <link rel="icon" href="/favicon.ico" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+      </Head>
+      <RCform formFields={formFields} />
```
Comment (Sing-Li) [inline app/pages/forms/formdemo.js:14]: Need to craft `RCform` as a re-usable component.   

Show us on this demo page.   Add 4 or 5  instances of different forms.

Similar to:
```
<h1>Sample of Form Component in Action</h1>
<Row>
<Col>
  <RCform  id="customersatisfaction" />
  </Col>
  <Col>
   <RCform id="survey1"/>
   </Col>
 </Row>
 <Row>
<Col>
  <RCform  id="mostlovedmusic" />
  </Col>
  <Col>
   <RCform id="COVID19Questionnaire"/>
   </Col>
 </Row>
 ```

**Review by Sing-Li — CHANGES_REQUESTED** [review]

**Comment by Dnouv** [comment]
A new demo page is added with a few preview forms, here is a screenshot of the page



Suggestions are most welcomed. Thank you!

**Review by Sing-Li — CHANGES_REQUESTED** [review]
Incredibly awesome work @Dnouv   🙏 

Please remove package-lock.json from the PR (perhaps add it your .gitignore)  and  it will be good enough for initial merge-down.   

Thanks!

**Comment by Dnouv** [comment]
Thank you so much for the feedback @Sing-Li .
I've reverted the changes of `package-lock.json` and pushed the changes.
Thank you!

**Pull request merged**